editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java

```diff
@@ -175,7 +170,10 @@ public void keyReleased(KeyEvent e) {
 			}
 		});
 
-		camelVersionCombo.addModifyListener(e -> validate());
+		camelVersionCombo.addModifyListener(e -> {
+			validate();
+			((FuseIntegrationProjectWizardTemplatePage)getWizard().getPage(Messages.newProjectWizardTemplatePageName)).refresh(camelVersionCombo.getText());
```

Comment:
I'm not a big fan of that but at least it avoids coupling the wizard pages too much with the filter
I think that the best would be to have an "Object Model" for the whole wizard which is shared between all pages and put listener on it but seems a bit complicated to implement.
What do you think?
Other suggestions?